extensions/kafka/src/main/java/com/hazelcast/jet/kafka/impl/StreamKafkaP.java

```diff
@@ -167,18 +174,38 @@ private void handleNewPartitions(int topicIndex, int newPartitionCount, boolean
         getLogger().info("New partition(s) assigned: " + newAssignments);
         eventTimeMapper.addPartitions(newAssignments.size());
         consumer.assign(currentAssignment.keySet());
-        if (oldTopicOffsets.length > 0 && !isRestoring) {
-            // For partitions detected later during the runtime we seek to their
-            // beginning. It can happen that a partition is added, and some messages
-            // are added to it before we start consuming from it. If we started at the
-            // current position, we will miss those, so we explicitly seek to the
-            // beginning.
-            getLogger().info("Seeking to the beginning of newly-discovered partitions: " + newAssignments);
-            consumer.seekToBeginning(newAssignments);
+        if (!isRestoring) {
+            if (oldTopicOffsets.length > 0) {
+                // For partitions detected later during the runtime we seek to their
+                // beginning. It can happen that a partition is added, and some messages
+                // are added to it before we start consuming from it. If we started at the
+                // current position, we will miss those, so we explicitly seek to the
+                // beginning.
+                getLogger().info("Seeking to the beginning of newly-discovered partitions: " + newAssignments);
+                consumer.seekToBeginning(newAssignments);
```

Comment:
you could have initial offset configured for a partition discovered later. I do not know if that makes much sense (probably not), but current implementation is not symmetric. If this is desired, it would be worthwhile to add a comment.